crates/cxx-qt-gen/test_outputs/qenum.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <cstdint>
+#include <cxx-qt-common/cxxqt_locking.h>
+#include <cxx-qt-common/cxxqt_maybelockguard.h>
+#include <cxx-qt-common/cxxqt_type.h>
+
+namespace cxx_qt::my_object {
+class MyObject;
+namespace cxx_qt_qenums {
```

Comment:
as discussed there might be some namespace changes?

Reply:
Done.

The enum now doesn't modify the namespace at all.
So the user get's full choice of the namespace the enum ends up in.

It now acts the same as declaring an enum in CXX normally would, but additionally registers the enum as a Q_ENUM inside the given QObject.
This seems to be the best way to keep compatibility with both CXX and Qt behavior.

As that may polute the global namespace with QEnum definitions, we should add support for the `#[namespace="..."]` attribute on qenums in a follow-up commit.
However, that requires fixing a few other bugs, including that `#[namespace="..."]` on a `#[qobject]` results in incorrect code generation in C++.